lsp-mode.el

```diff
@@ -7081,22 +7109,24 @@ When prefix UPDATE? is t force installation even if the server is present."
 
 (defun lsp-async-start-process (callback error-callback &rest command)
   "Start async process COMMAND with CALLBACK and ERROR-CALLBACK."
-  (make-process
-   :name (cl-first command)
-   :command command
-   :sentinel (lambda (proc _)
-               (when (eq 'exit (process-status proc))
-                 (if (zerop (process-exit-status proc))
-                     (condition-case err
-                         (funcall callback)
-                       (error
-                        (funcall error-callback (error-message-string err))))
-                   (display-buffer " *lsp-install*")
-                   (funcall error-callback
-                            (format "Async process '%s' failed with exit code %d"
-                                    (process-name proc) (process-exit-status proc))))))
-   :buffer " *lsp-install*"
-   :noquery t))
+  (let ((name (cl-first command)))
+    (with-current-buffer (compilation-start (combine-and-quote-strings command) t
```

Comment:
Is this `combine-and-quote-string` works on Windows? Its document is saying
>Note that this is not intended to protect STRINGS from
interpretation by shells, use ‘shell-quote-argument’ for that.

Reply:
It indeed doesn't; I am currently refactoring the powershell installation script to use a single lsp-async-start-process instead, and this caused breakage. I will change it to `(mapconcat #'shell-quote-argument args " ")` soon.